implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OtlpCumulativeMeterRegistryTest.java

```diff
@@ -91,11 +91,11 @@ void timer() {
         timer.record(111, TimeUnit.MILLISECONDS);
         clock.add(otlpConfig().step());
         timer.record(4, TimeUnit.MILLISECONDS);
-        assertThat(writeToMetric(timer).toString()).isEqualTo(
-                "name: \"web.requests\"\n" + "description: \"timing web requests\"\n" + "unit: \"milliseconds\"\n"
-                        + "histogram {\n" + "  data_points {\n" + "    start_time_unix_nano: 1000000\n"
-                        + "    time_unix_nano: 60001000000\n" + "    count: 4\n" + "    sum: 202.0\n" + "  }\n"
-                        + "  aggregation_temporality: AGGREGATION_TEMPORALITY_CUMULATIVE\n" + "}\n");
+        assertThat(writeToMetric(timer).toString()).isEqualTo("name: \"web.requests\"\n"
```

Comment:
I guess if we are making a change at the lower layer(in this case OTEL protocol). It would be great to add tests that cover both Cumulative and Delta cases.
@jack-berg Do you mind if I add additional commits with tests?

Reply:
Please do!

Reply:
I just realized that I wouldn't have permission. But that's not a big deal, adding things to OtlpMeterRegistryTest#assertHistogram to assert the bucket list is always non-zero.

The ideal way to do it should be to assert that the sum of values in the buckets_count list is equal to the count. But due to [this open issue](https://github.com/micrometer-metrics/micrometer/issues/3934), the numbers won't match up when values fall into Positive Infinity. So for now asserting that bucketCounts is always Positive would be a good start.

Reply:
The linked issue is now fixed, which means this PR will need to be rebased to resolve conflicts.